src/str.js

```diff
@@ -999,6 +1087,20 @@ Sk.builtin.str.prototype["istitle"] = new Sk.builtin.func(function (self) {
     return new Sk.builtin.bool( cased);
 });
 
+Sk.builtin.str.prototype["encode"] = new Sk.builtin.func(function (self, encoding, errors) {
+    Sk.builtin.pyCheckArgsLen("encode", arguments.length, 1, 3);
+    encoding = encoding || Sk.builtin.str.$utf8;
+    Sk.builtin.pyCheckType("encoding", "string", Sk.builtin.checkString(encoding));
+    encoding = encoding.v;
+    if (errors !== undefined) {
+        Sk.builtin.pyCheckType("errors", "string", Sk.builtin.checkString(errors));
+        errors = errors.v;
+    } else {
+        errors = "strict";
+    }
+    return Sk.builtin.bytes.$strEncode(self, encoding, errors);
```

Comment:
In Python 2 mode, we need to return a string here (ie `new Sk.builtin.str(<value>.$jsstr())`)

Reply:
have resolved this this with a `Sk.__future__` check

Reply:
as above. some py2 tests here might be desirable... i'm not sure the `__future__` flag is sufficient or if the `$jsstr()` does what it's supposed to...